sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -921,17 +939,29 @@ def get_throughput(self, **kwargs: Any) -> ThroughputProperties:
         :rtype: ~azure.cosmos.ThroughputProperties
         """
         response_hook = kwargs.pop('response_hook', None)
-        properties = self._get_properties()
-        link = properties["_self"]
-        query_spec = {
-            "query": "SELECT * FROM root r WHERE r.resource=@link",
-            "parameters": [{"name": "@link", "value": link}],
-        }
-        throughput_properties = list(self.client_connection.QueryOffers(query_spec, **kwargs))
-        if not throughput_properties:
-            raise CosmosResourceNotFoundError(
-                status_code=StatusCodes.NOT_FOUND,
-                message="Could not find ThroughputProperties for container " + self.container_link)
+        throughput_properties: List[Dict[str, Any]]
+        # In case of a container recreate we will run this a second time
+        for attempt in range(2):
+            if attempt == 1:
+                # Refresh cache if no offer throughput was received the first time
+                self.read()
+            properties = self._get_properties()
+            link = properties["_self"]
+            query_spec = {
+                "query": "SELECT * FROM root r WHERE r.resource=@link",
+                "parameters": [{"name": "@link", "value": link}],
+            }
+            options: Mapping[str, Any] = {}
+
+            throughput_properties = list(self.client_connection.QueryOffers(query_spec, options, **kwargs))
+            if not throughput_properties:
+                # Try one more time after refreshing the container properties cache
+                if attempt < 1:
+                    continue
+                raise CosmosResourceNotFoundError(
```

Comment:
Goes back to a point @ealsur made about mapping the appropriate sub-status code here?

Reply:
Yes, I made a python specific client side substatus code for this and it is now raised in retry utils.